config/v1alpha1/types_image_policy.go

```diff
@@ -143,6 +151,39 @@ type PolicyFulcioSubject struct {
 	SignedEmail string `json:"signedEmail"`
 }
 
+// PKI defines the root of trust based on Root CA(s) and corresponding intermediate certificates.
+// +openshift:enable:FeatureGate=SigstoreImageVerificationPKI
+type PKI struct {
+	// caRootsData contains base64-encoded data of a certificate bundle PEM file, which contains one or more CA roots in the PEM format.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=8192
+	CertificateAuthorityRootsData []byte `json:"caRootsData"`
+	// caIntermediatesData contains base64-encoded data of a certificate bundle PEM file, which contains one or more intermediate certificates in the PEM format.
+	// caIntermediatesData requires CertificateAuthorityRoots to be set.
+	// +optional
+	// +kubebuilder:validation:MaxLength=8192
+	CertificateAuthorityIntermediatesData []byte `json:"caIntermediatesData,omitempty"`
+
+	// pkiCertificateSubject defines the requirements imposed on the subject to which the certificate was issued.
+	// +kubebuilder:validation:Required
+	// +required
+	PKICertificateSubject *PKICertificateSubject `json:"pkiCertificateSubject"`
+}
+
+// PKICertificateSubject defines the requirements imposed on the subject to which the certificate was issued.
+// +kubebuilder:validation:XValidation:rule="has(self.email) || has(self.hostname)", message="at least one of email or hostname must be set in pkiCertificateSubject"
+// +openshift:enable:FeatureGate=SigstoreImageVerificationPKI
+type PKICertificateSubject struct {
+	// email specifies the expected email address imposed on the subject to which the certificate was issued.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^\\S+@\\S+$')`,message="invalid email address in pkiCertificateSubject"
+	Email string `json:"email,omitempty"`
+	// hostname specifies the expected hostname imposed on the subject to which the certificate was issued.
+	// +optional
+	// +kubebuilder:validation:XValidation:rule=`self.matches('^(\\*\\.)?([a-zA-Z0-9-]+\\.)*[a-zA-Z0-9-]+\\.[a-zA-Z]{2,}$')`,message="invalid hostname in pkiCertificateSubject"
```

Comment:
What is this regex trying to check against? Where did it come from? Checking that this is a valid domain/subdomain? Would using the format library to check for DNS 1123 subdomain name be sufficient here? Though that wouldn't allow a wildcard optionally preceding 🤔 We can get round that I'm sure
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.startsWith('*.') ? !format.dns1123subdomain().validate(self.replace('*.', '', 1)).hasValue() : !format.dns1123subdomain().validate(self).hasValue()",message="hostname should be a valid dns 1123 subdomain name, optionally prefixed by '*.'. It should consist only of lowercase alphanumeric characters, hyphens, periods and the optional preceding asterisk."
```

Reply:
@JoelSpeed do you know how we can use the format library here? it failed https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_api/2088/pull-ci-openshift-api-master-verify-crd-schema/1864499295143596032

Reply:
There's an example of it working in https://github.com/openshift/api/blob/c1fdeb0788c16659238d93ec45ce2e798c14eb88/machine/v1/types_controlplanemachineset.go#L55

I think we have typos in the code here, sorry, should be `dns1123Subdomain()` and not `dns1123subdomain()`